features/ui/button/button.tsx

```diff
@@ -1,9 +1,80 @@
 import { ButtonHTMLAttributes } from "react";
 import classNames from "classnames";
 import styles from "./button.module.scss";
+import Image from "next/image";
 
-export function Button(props: ButtonHTMLAttributes<HTMLButtonElement>) {
+export enum Size {
+  small = "small",
+  medium = "medium",
+  large = "large",
+  xlarge = "xlarge",
+}
+
+export enum Color {
+  primary = "primary",
+  secondary = "secondary",
+  gray = "gray",
+  empty = "empty",
+  emptyGray = "emptyGray",
+  error = "error",
+}
+
+export interface ButtonIcon {
+  src: string;
+  alt: string;
+  width: number;
+  height: number;
+}
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  size?: Size;
+  color?: Color;
+  isDisabled?: boolean;
+  iconLeft?: ButtonIcon;
+  iconRight?: ButtonIcon;
+  text?: string;
```

Comment:
An interesting alternative could be to use component composition here. So you could add a separate `ButtonIcon` component and then render the button like

```
<Button>
  <ButtonIcon src="..." />
  Click me
</Button>
```

Reply:
Oh! Yeah, I'm a fan of that. The props are already annoying! I don't even think I'd need to make a ` <ButtonIcon ... />` component as I started out with making the styling work with just adding an `<Image .. />` component a la
```
    <Button
      onClick={() => alert("Clicked!")}
      size={size}
      color={color}
      disabled={disabled}
      noStyles={noStyles}
    >
      <Image src="/icons/circle.svg" width={20} height={20} alt="Left icon" />
      Button UI
      <Image src="/icons/circle.svg" width={20} height={20} alt="Right icon" />
    </Button>
```
...but, for some reason, I really wanted to add a `label`/`text` prop to the component and to get the images/icons to to position properly I then added the icon props. I dunno. A quick refactor wouldn't be out of the question, though, and it'd probably be a little more sensible in the long run.